src/presign/participant.rs

```diff
@@ -192,11 +237,99 @@ impl PresignParticipant {
             ));
         }
 
-        Ok(ret_messages)
+        let mut broadcast_messages = self.broadcast(
+            rng,
+            &MessageType::Presign(PresignMessageType::RoundOneBroadcast),
+            serialize!(&r1_public_broadcast)?,
+            message.id(),
+            "PresignR1Ciphertexts",
+        )?;
+        out_messages.append(&mut broadcast_messages);
+
+        // Additionally, handle any round 1 messages which may have been received too
+        // early
+        let retrieved_messages = self.fetch_messages(
+            MessageType::Presign(PresignMessageType::RoundOne),
+            message.id(),
+        )?;
+        let mut presign_record = None;
+        for msg in retrieved_messages.iter() {
+            let (pr, mut r2_msg) = self.handle_round_one_msg(rng, msg, main_storage)?;
+            out_messages.append(&mut r2_msg);
+            if pr.is_some() {
+                presign_record = pr;
+            }
```

Comment:
Here, we assign to `presign_record` possibly many times (since we are in a loop). Only the assignment will be returned from this function. All previous assignments will get overwritten by the most recent one. Is this the intended behavior? If so, can you add a comment making this explicit?